### PR TITLE
Editoast: Invalidate cache without chartos

### LINKED ISSUE #2894: Invalidate layers from editoast using redis connection and remove `reqwest` dependency
Thanks to https://github.com/DGEXSolutions/osrd/commit/850952e8ad2ac3ed7a32d8c73bbdd1b896650dff editoast has the possibility to connect to the redis database and therefore invalidate caches without using chartos http endpoint

# Definition of done 

- [ ] Check the possibility of an integration test (must put data in the redis and check if invalidations actually remove data)
- [ ] If above is not possible, move on the map or create an infra to see if layers are invalidated
- [ ] Remove `reqwest` dependency

### PR DESCRIPTION
**Pull request by bloussou**
This pr is : 
- Invalidating cache using rust redis client
- Removing `reqwest` dependency
- Removing the need to connect to chartos from editoast

close #2894

### CODE REVIEW

File: editoast/src/chartos/layer_cache.rs [hunk 1]
```diff
@@ -92,13 +84,31 @@ pub fn count_tiles(max_zoom: u64, bbox: &BoundingBox) -> u64 {
     count
 }
 
+pub fn get_layer_cache_prefix(layer_name: &str, infra_id: i32) -> String {
+    format!("chartis.layer.{layer_name}.infra_{infra_id}")
```
Comment (flomonster) [inline editoast/src/chartos/layer_cache.rs:89]:
```suggestion
    format!("editoast.layer.{layer_name}.infra_{infra_id}")
```

Or `chartos` but I think we should stop using this name.

File: editoast/src/main.rs [hunk 2]
```diff
@@ -153,7 +148,14 @@ async fn generate(
...
         if infra.refresh(&mut conn, args.force, &infra_cache)? {
-            chartos::invalidate_all(infra.id, &chartos_config).await;
+            let cfg = RedisPoolConfig::from_url(&redis_config.redis_url);
+            let pool = cfg.create_pool(Some(Runtime::Tokio1)).unwrap();
+            chartos::invalidate_all(
+                &RedisPool(pool),
+                &MapLayers::parse().layers.keys().cloned().collect(),
+                infra.id,
+            )
+            .await;
```
Comment (flomonster) [inline editoast/src/main.rs]: This could be executed when we clear the generated data of an infra.

File: editoast/src/views/infra/edition.rs [hunk 3]
```diff
@@ -39,7 +39,13 @@ async fn edit<'a>(
         })
         .await?;
 
-    chartos::invalidate_zone(infra, chartos_config, &invalid_zone).await;
+    chartos::invalidate_zone(
```
Comment (flomonster) [inline editoast/src/views/infra/edition.rs:44]: Can you add a TODO and/or create an issue about invalidate only layers that have been updated.

File: editoast/src/chartos/mod.rs [hunk 4]
```diff
@@ -1,102 +1,139 @@
...
+/// * `infra_id` - Infra to on which the layer must be invalidated
+/// * `layer_name` - Layer on which invalidation must be done
+/// * `zone` - Zone to invalidate
+async fn invalidate_layer_zone(
+    redis_pool: &RedisPool,
+    infra_id: i32,
+    layer_name: &str,
+    zone: &InvalidationZone,
+) -> Result<(), RedisError> {
+    let max_tiles = 120;
```
Comment (flomonster) [inline editoast/src/chartos/mod.rs]: Can we add this in the the configuration file (or as a cli arg)?
Also the default value was set to `250 000`.

**Review by flomonster — CHANGES_REQUESTED** [review]
Great PR !

Can you adapt the docker-compose files and CI. Since editoast doesn't need chartos to work but redis we can simplify things.

File: editoast/src/chartos/layer_cache.rs (hunk 1 above)
Comment (bloussou) [reply editoast/src/chartos/layer_cache.rs:89]: I will do that once chartos will not be responsibe to cache tiles in the redis (I have to do #2893)

File: editoast/src/views/infra/edition.rs (hunk 3 above)
Comment (bloussou) [reply editoast/src/views/infra/edition.rs:44]: Yes : https://github.com/DGEXSolutions/osrd/issues/2955

**Review by flomonster — DISMISSED** [review]

**Review by flomonster — APPROVED** [review]

**Pull request merged**